test/Dialect/FIRRTL/lower-layers.mlir

```diff
@@ -1,11 +1,273 @@
 // RUN: circt-opt -firrtl-lower-layers -split-input-file %s | FileCheck %s
 
+firrtl.circuit "Test" {
+  firrtl.module @Test() {}
+
+  firrtl.layer @A bind {
+    firrtl.layer @B bind {
+      firrtl.layer @C bind {}
+    }
+  }
+  firrtl.layer @B bind {}
+
+  firrtl.extmodule @Foo(out o : !firrtl.probe<uint<1>, @A>)
+
+  //===--------------------------------------------------------------------===//
```

Comment:
:heart_eyes: :open_mouth: :heart_eyes_cat:  thanks for the (nicely organized!) tests! :rocket: